### PR TITLE
fix(docker): build the server image for the host architecture and resolve Drogon's static deps (#4546)

### LINKED ISSUE #4546: Dockerfile.server: arm64 image build unreachable (no TRIPLET passed) and drogon canary link fails (no -Dpkg_config_path)
Two separate defects in the HEAD-image build path, found while trying to run the full `/test` gate on an Apple Silicon Mac. The first masks the second.

## 1. The `/test` harness never passes `TRIPLET`, so Apple Silicon always cross-builds x64 and dies in vcpkg

`deploy/docker/Dockerfile.server:34` declares `ARG TRIPLET=x64-linux` and its own comment says *"Pass `--build-arg TRIPLET=arm64-linux` for arm64"*. `.claude/skills/test/SKILL.md` (Phase 1, "Build (HEAD docker images)") invokes `docker build ... -f deploy/docker/Dockerfile.server .` with **no** `--build-arg TRIPLET` and no `--platform`.

On an arm64 host this cross-builds the x64 triplet under emulation and fails inside vcpkg:

```
error: vcpkg was unable to detect the active compiler's information
```
(`detect_compiler`, `vcpkg_execute_required_process.cmake:127`)

This is long-standing, not new: a byte-identical signature appears in `~/.local/share/yuzu/test-runs/1788861795-13146/build-images.log` for the unrelated branch `po/admin-plane` on 2026-09-08. The effect is that Phase 1's image gate and the Phase 2 upgrade test are **red on every Apple Silicon Mac for every branch, including `dev`**, so a literal `overall_status=PASS` full `/test` is unobtainable on that hardware.

**Verified fix for this half:** `docker build --platform linux/arm64 --build-arg TRIPLET=arm64-linux -f deploy/docker/Dockerfile.server .` gets the vcpkg stage to complete cleanly — `arm64-linux` installs fine. The harness should select the triplet from the host arch rather than defaulting to x64.

## 2. With the vcpkg stage unblocked, the drogon canary fails to link — the Dockerfile omits a documented load-bearing flag

Once past vcpkg, the build reaches meson and fails:

```
FAILED: server/core/yuzu_drogon_canary
HttpRequestImpl.cc: undefined reference to `BrotliDecoderCreateInstance'
Utilities.cc: undefined reference to `uuid_generate'
Utilities.cc: undefined reference to `deflateInit2_' / `compressBound' / `inflate' ...
```

`server/core/meson.build:753-759` resolves drogon's transitive statics (`libcares`, `zlib`, `uuid`, `libbrotlienc`, `libbrotlidec`) **via pkg-config with `required: false`**, on the stated assumption that "All resolve via pkg-config in the vcpkg tree on Linux".

But `deploy/docker/Dockerfile.server:86` passes only `-Dcmake_prefix_path=/src/vcpkg_installed/${TRIPLET}` — there is no `-Dpkg_config_path` and no `PKG_CONFIG_PATH` in the image. `CLAUDE.md` is explicit that this is not optional:

> **A manual `meson setup` needs two load-bearing flags, both pointing INTO THE TREE** — `-Dcmake_prefix_path` *and* `-Dpkg_config_path` against a populated `vcpkg_installed/<triplet>`. Neither is optional and neither substitutes for the other.

So the five lookups silently return not-found, `required: false` drops them, and the static drogon link fails with exactly those symbols. Because defect 1 kills the build earlier on arm64, and CI only builds x64, this has never been observed.

**Proposed fix:** add `-Dpkg_config_path=/src/vcpkg_installed/${TRIPLET}/lib/pkgconfig` to the Dockerfile's `meson setup`. Consider also whether `required: false` on those five is right at all — a silent drop is what turned a missing flag into a confusing link error rather than a clear configure failure.

## Why this matters beyond one laptop
arm64 Linux is a shipping target (arm64 release builds return natively via the macOS runner), and the drogon canary exists specifically to catch #375-class link regressions. It cannot catch them on a platform where it has never successfully linked.

Found while gating the macOS Spark/Reflex/DEX/Guardian programme (#4513).

### PR DESCRIPTION
**Pull request by Tr3kkR**
Fixes #4546. Two independent defects that together made the arm64 server image unbuildable.

## 1. The image build never received a triplet

`Dockerfile.server` accepts `--build-arg TRIPLET` and defaults to `x64-linux`. Nothing in the test
harness passed it, so on an arm64 host the build attempted a cross-architecture vcpkg install and
died at compiler detection. The harness now derives both the Docker platform and the vcpkg triplet
from the host's own architecture and passes them through.

## 2. Drogon's transitive static dependencies resolved to nothing

`server/core/meson.build` resolves Drogon's transitive statics — c-ares, zlib, uuid, and both Brotli
halves — through pkg-config with `required: false`. Without `-Dpkg_config_path` pointing into the
populated vcpkg tree, all five silently resolved to not-found and the link failed with undefined
references. **The `required: false` is what makes this quiet**: there is no configure-time error, only
a link-time failure much later with no indication of the cause.

Both flags are now passed in all four Dockerfiles, matching the rule already stated in `CLAUDE.md`
that a manual `meson setup` needs `-Dcmake_prefix_path` **and** `-Dpkg_config_path`, both pointing
into the tree, and that neither substitutes for the other.

## Proof

The arm64 image builds end to end: exit 0, zero `FAILED:` lines, zero undefined references,
`[284/304] Linking target server/core/yuzu_drogon_canary`, producing a 247 MB image containing a
62,215,416-byte `yuzu-server`.

Native darwin-arm64 is separately confirmed for all three components: `yuzu-server` and `yuzu-agent`
as Mach-O arm64 binaries, and the gateway as `yuzu_gw-0.2.0` on ERTS 16.4 with an arm64 `beam.smp`.

## What this does not fix

Cross-architecture x64-linux image builds on an arm64 host still fail at vcpkg compiler detection
under emulation. That is a separate, pre-existing, branch-independent condition, reproduced on an
unrelated branch on 2026-09-08. This change makes the **host-native** image build work, which is what
the local test gate actually needs.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by Tr3kkR** [comment]
**The macOS debug failure is not caused by this PR.** The agent unit-test binary was killed by `SIGSEGV` after every test case had started and finished — every `[progress] START` line has a matching `END` (run 35328248154). This PR changes only the four Dockerfiles, `.claude/skills/test/SKILL.md` and a changelog fragment, none of which is compiled into that binary, so the crash cannot come from this diff. The most likely cause is the open intermittent macOS crash #4488 in `run_bounded_subprocess()`, and `dev`'s own CI has been failing intermittently. The failed job has been re-run.

File: .claude/skills/test/SKILL.md [hunk 1]
```diff
@@ -197,16 +197,31 @@ subprojects define the same name.
...
+# detect the active compiler's information" (detect_compiler) — the image gate
+# then fails on EVERY branch on Apple Silicon, `dev` included, and cascades to
+# the Phase 2 upgrade test. Derive it from `uname -m` and pass --platform to
+# match, so an arm64 box builds arm64-linux natively and an x64 box is
+# byte-unchanged from the historical behaviour.
 if [[ "$MODE" != "quick" ]]; then
     (
         set -e
         BUILD_START=$(date +%s)
+        case "$(uname -m)" in
```
Comment (FortitudeEtc) [inline .claude/skills/test/SKILL.md:213]: LOW: `uname -m` reflects the CLI host, not the active Docker builder/context — a remote or translated Docker context could still select a mismatched platform. Matches existing repo precedent (`scripts/start-demo.sh` does the same uname-only mapping); `scripts/start-viz-uat.sh` shows a nicer pattern (an env-var override, e.g. `VIZ_UAT_PLATFORM`). Not blocking — edge case, not the scenario #4546 was about.

Comment (FortitudeEtc) [inline .claude/skills/test/SKILL.md:213]: LOW: no CI leg exercises this new arm64 branch (self-hosted Linux CI is x64-only; existing arm64 QEMU image builds don't run this exact host-detection code path). Failure mode would be loud and local to arm64 dev boxes, not silent. Not blocking — optional follow-up if an arm64 CI leg is ever justified.

**Review by FortitudeEtc — APPROVED** [review]
## /go review — PR #4551

Static-first review (Kimi + Codex, independent then cross-examined) followed by real dynamic evidence: I built `Dockerfile.server` and the `Dockerfile.server.chisel` builder stage at this head on an x64 host — both linked all 304/304 targets and produced a running binary (`yuzu-server --version` → `0.13.1+0`). The new `-Dpkg_config_path` line is exactly the second half of the CLAUDE.md "two load-bearing flags" invariant, and it's now correctly present at all four Dockerfile sites (`${TRIPLET}` and `$T` conventions both verified). The new `case "$(uname -m)")` mapping in `.claude/skills/test/SKILL.md` resolves correctly on x64 (byte-unchanged claim holds); I don't have arm64 hardware to reproduce the native-host scenario directly, which is the one gap inherent to reviewing an arch-specific fix from a single-arch box.

No CRITICAL/HIGH found by either reviewer or by me. **Approving.** Two non-blocking MEDIUM completeness gaps worth your judgment call (both outside this diff's own correctness — the fix itself is right):

1. **`.codex/skills/test/SKILL.md` doesn't carry the fix.** Its Phase 1 description is still generic ("docker images... except in quick mode"), so a Codex-driven `/test` on Apple Silicon has no documented triplet/platform pairing rule and could reconstruct the same unqualified build #4546 was about. Not a regression, just unfixed on that leg.
2. **`scripts/docker-start-UAT.sh`'s `rebuild` subcommand still builds unqualified** (line 221: `docker build -t yuzu-server -f ".../Dockerfile.server" "$YUZU_ROOT"`, no `--platform`/`--build-arg TRIPLET`). Same root cause survives at a second, actively-used (allowlisted in `.claude/settings.json`) entry point. Outside this PR's stated `/test`-gate scope, but worth a follow-up.

Two LOW items inline below. Also: both reviewers independently flagged, then withdrew, the changelog fragment being named `4546-...` (issue #) instead of `4551-...` (PR #) — `changelog.d/README.md` explicitly permits either and the lint passes, so that's a non-issue.

Full review trail (STATIC/DYNAMIC/cross-exam transcripts) archived at `/tmp/yuzu-go-pr4551` on the reviewing host.
